pkg/storage/stores/tsdb/store.go

```diff
@@ -62,35 +65,43 @@ func NewStore(
 		logger:            logger,
 	}
 
-	if err := storeInstance.init(name, indexShipperCfg, schemaCfg, objectClient, limits, tableRange, reg); err != nil {
+	if err := storeInstance.init(name, indexShipperCfg, schemaCfg, objectClient, limits, tableRange, reg, idxCache); err != nil {
 		return nil, nil, err
 	}
 
 	return storeInstance, storeInstance.Stop, nil
 }
 
-func (s *store) init(name string, indexShipperCfg indexshipper.Config, schemaCfg config.SchemaConfig, objectClient client.ObjectClient,
-	limits downloads.Limits, tableRange config.TableRange, reg prometheus.Registerer) error {
+func (s *store) init(name string, indexCfg IndexCfg, schemaCfg config.SchemaConfig, objectClient client.ObjectClient,
+	limits downloads.Limits, tableRange config.TableRange, reg prometheus.Registerer, idxCache cache.Cache) error {
+
+	sharedCacheClient = idxCache
+
+	usePostingsCache := indexCfg.CachePostings && indexCfg.Mode == indexshipper.ModeReadOnly && idxCache != nil
+	openFn := func(p string) (indexshipper_index.Index, error) {
+		return OpenShippableTSDB(p, IndexOpts{UsePostingsCache: usePostingsCache})
+	}
```

Comment:
Another idea to avoid the module scoped cache instance: Could we pass the `idxCache` instead of `usePostingsCache`?

```suggestion
func (s *store) init(name string, indexCfg IndexCfg, schemaCfg config.SchemaConfig, objectClient client.ObjectClient,
	limits downloads.Limits, tableRange config.TableRange, reg prometheus.Registerer, idxCache cache.Cache) error {

	var cache.Cache sharedCache
	if indexCfg.CachePostings && indexCfg.Mode == indexshipper.ModeReadOnly && idxCache != nil {
	    sharedCache = idxCache
	}
	openFn := func(p string) (indexshipper_index.Index, error) {
		return OpenShippableTSDB(p, IndexOpts{UsePostingsCache: sharedCache})
	}
```